File: smartHome/app/src/main/java/com/sunasterisk/smarthomejava/MainActivity.java
package com.sunasterisk.smarthomejava;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;

import com.sunasterisk.smarthomejava.adapter.AdapterLed;
import com.sunasterisk.smarthomejava.model.Led;

import org.eclipse.paho.android.service.MqttAndroidClient;
import org.eclipse.paho.client.mqttv3.IMqttActionListener;
import org.eclipse.paho.client.mqttv3.IMqttDeliveryToken;
import org.eclipse.paho.client.mqttv3.IMqttToken;
import org.eclipse.paho.client.mqttv3.MqttCallback;
import org.eclipse.paho.client.mqttv3.MqttClient;
import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
import org.eclipse.paho.client.mqttv3.MqttException;
import org.eclipse.paho.client.mqttv3.MqttMessage;

import java.io.UnsupportedEncodingException;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    RecyclerView recyclerView;
    List<Led> leds;
    RecyclerView.Adapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Intent mService = new Intent(this, MainService.class);
        startForegroundService(mService);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            startForegroundService(mService);
        } else {
            startService(mService);
        }
        recyclerView = findViewById(R.id.recyclerViewContains);

        leds = new ArrayList<Led>();
        //Tự phát sinh 50 dữ liệu mẫu
        for (int i = 1; i <= 50; i++) {
            leds.add(new Led(i , i%2));
        }

        adapter = new AdapterLed(leds, this);

        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);

        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(linearLayoutManager);

    }
}

File: RemoteSmartHouse/app/src/main/java/com/sunasterisk/remotesmarthouse/MainActivity.kt
package com.sunasterisk.remotesmarthouse

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.google.gson.Gson
import com.sunasterisk.remotesmarthouse.model.Air
import com.sunasterisk.remotesmarthouse.model.LED
import com.sunasterisk.remotesmarthouse.model.RFID
import com.sunasterisk.remotesmarthouse.model.Sensors
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var airs = listOf<Air>(Air(1,value = 20))
        var leds = listOf<LED>(LED(1,value = 20))
        var RFIDs = listOf<RFID>(RFID(1))
        var s = Sensors(leds,airs,RFIDs)
        var gson = Gson()
        text.text =    gson.toJson(s)
    }
}
